### PR TITLE
Changes to support using posix test suite on SplitFS

### PR DESCRIPTION
**Pull request by OmSaran**
1. Delegate the handling of STDIO files (stdin, stdout, stderr) to posix.
2. Use the originally initialised fileop handlers instead of setting it to NULL when a file is closed.
3. Use posix system call to truncate the SHM file in SplitFS' execvp implementation (similar to execve). 
4. Use PID for the shared memory regions as mentioned in paper.
5. Honour file descriptors passing on exec without needing to compile with workload_tar flag.

### CODE REVIEW

**Comment by OmSaran** [comment]
1. Add test files
2. Add Makefile to run the tests

Makefile usage:
To run all tests:
make

To run individual tests:
cd tests
make pjd.<mode> -- where mode is oneof: (posix,sync,strict)

File: splitfs/fileops_hub.c [hunk 1]
```diff
@@ -815,6 +812,8 @@ RETT_OPENAT _hub_OPENAT(INTF_OPENAT)
 }
 
 RETT_EXECVE _hub_EXECVE(INTF_EXECVE) {
+	int pid = getpid();
+	char buf[BUF_SIZE];
```
Comment (rohankadekodi) [inline splitfs/fileops_hub.c:816]: Can we name this something else? Probably exec-hub-filename?

Comment (rohankadekodi) [inline splitfs/fileops_hub.c:816]: Can we rename this to something else? Like exec-hub-filename

File: splitfs/fileops_hub.c [hunk 2]
```diff
@@ -832,7 +831,8 @@ RETT_EXECVE _hub_EXECVE(INTF_EXECVE) {
 			hub_ops[i] = 2;
 	}
 
-        exec_hub_fd = shm_open("exec-hub", O_RDWR | O_CREAT, S_IRUSR | S_IWUSR);
+	sprintf(buf, "exec-hub-%d", pid);
```
Comment (rohankadekodi) [inline splitfs/fileops_hub.c:834]: Can you fix the indentation?

File: splitfs/fileops_hub.c [hunk 3]
```diff
@@ -1101,16 +1107,18 @@ RETT_CLOSE _hub_CLOSE(INTF_CLOSE)
 	DEBUG("_hub_CLOSE is calling %s->CLOSE\n", _hub_fd_lookup[file]->name);
 	
 	struct Fileops_p* temp = _hub_fd_lookup[file];
-	_hub_fd_lookup[file] = NULL;
+
+  // Restore it to the state similar to the initialised state.
+  _hub_fd_lookup[file] = _hub_fileops;
```
Comment (rohankadekodi) [inline splitfs/fileops_hub.c:1112]: Can you fix the indentation?

File: splitfs/fileops_nvp.c [hunk 4]
```diff
@@ -1021,8 +1021,11 @@ void _nvp_SHM_COPY() {
         int exec_ledger_fd = -1;
 	int i,j;
 	unsigned long offset_in_map = 0;
-	
-	exec_ledger_fd = shm_open("exec-ledger", O_RDONLY, 0666);
+	int pid = getpid();
+	char buf[BUF_SIZE];
```
Comment (rohankadekodi) [inline splitfs/fileops_nvp.c:1025]: Can you rename this to something else? Like exec-nvp-filename

File: splitfs/fileops_nvp.c [hunk 5]
```diff
@@ -5911,6 +5916,9 @@ RETT_DUP2 _nvp_DUP2(INTF_DUP2)
 	nvf2->node = nvf->node;
 	nvf2->valid = nvf->valid;
 	nvf2->posix = nvf->posix;
+  // Increment the refernce count as this file 
+  // descriptor is pointing to the same NVFNode
+  nvf2->node->reference++;
```
Comment (rohankadekodi) [inline splitfs/fileops_nvp.c:5921]: Can you fix the indentation?

File: splitfs/fileops_hub.c (hunk 1 above)
Comment (OmSaran) [reply splitfs/fileops_hub.c]: Sure Rohan, done.

File: splitfs/fileops_hub.c (hunk 2 above)
Comment (OmSaran) [reply splitfs/fileops_hub.c]: Sure, done.

File: splitfs/fileops_hub.c (hunk 3 above)
Comment (OmSaran) [reply splitfs/fileops_hub.c]: Sure, done.

File: splitfs/fileops_nvp.c (hunk 4 above)
Comment (OmSaran) [reply splitfs/fileops_nvp.c]: Sure, done.

File: splitfs/fileops_nvp.c (hunk 5 above)
Comment (OmSaran) [reply splitfs/fileops_nvp.c]: Sure, done.

File: README.md [hunk 6]
```diff
@@ -100,6 +100,16 @@ SplitFS is under active development.
 1. The current implementation of SplitFS handles the following system calls: `open, openat, close, read, pread64, write, pwrite64, fsync, unlink, ftruncate, fallocate, stat, fstat, lstat, dup, dup2, execve and clone`. The rest of the calls are passed through to the kernel.
 2. The current implementation of SplitFS works correctly for the following applictions: `LevelDB running YCSB, SQLite running TPCC, tar, git, rsync`. This limitation is purely due to the state of the implementation, and we aim to increase the coverage of applications by supporting more system calls in the future.
 
+## Testing
+[PJD POSIX Test Suite](https://www.tuxera.com/community/posix-test-suite/) that tests primarily the metadata operations was run on SplitFS and yielded the following result.
```
Comment (vijay03) [inline README.md:104]: Can we add some details on how to run the tests?

Comment (OmSaran) [reply README.md:104]: Sure Vijay, added

**Pull request merged**